plugins/http_plugin/http_plugin.cpp

```diff
@@ -287,13 +328,201 @@ namespace eosio {
+            abstract_conn_impl& operator=(abstract_conn_impl&&) noexcept = default;
+
+            bool verify_max_bytes_in_flight() override {
+               return _impl.verify_max_bytes_in_flight(_conn);
+            }
+
+            void handle_exception()override {
+               http_plugin_impl::handle_exception<T>(_conn);
+               _conn->send_http_response();
+            }
+
+            detail::connection_ptr<T> _conn;
+            http_plugin_impl &_impl;
+         };
+
+         /**
+          * Helper to construct an abstract_conn_impl for a given connection and instance of http_plugin_impl
+          * @tparam T - The downstream parameter for the connection_ptr
+          * @param conn - existing connection_ptr<T>
+          * @param impl - reference to the ownint http_plugin_impl
+          * @return abstract_conn_ptr backed by type specific implementations of the methods
+          */
+         template<typename T>
+         static detail::abstract_conn_ptr make_abstract_conn_ptr( detail::connection_ptr<T> conn, http_plugin_impl& impl ) {
+            return std::make_shared<abstract_conn_impl<T>>(conn, impl);
+         }
+
+         /**
+          * Helper type that wraps an object of type T and records its "in flight" size to
+          * http_plugin_impl::bytes_in_flight using RAII semantics
+          *
+          * @tparam T - the contained Type
+          */
+         template<typename T>
+         struct in_flight {
+            in_flight(T&& object, http_plugin_impl& impl)
+            :object(std::move(object))
+            ,impl(impl)
+            {
+               count = detail::in_flight_sizeof(object);
+               impl.bytes_in_flight += count;
+            }
+
+            ~in_flight() {
+               if (count) {
+                  impl.bytes_in_flight -= count;
+               }
+            }
+
+            // No copy constructor, but allow move
+            in_flight(const in_flight&) = delete;
+            in_flight(in_flight&& from)
+            :object(std::move(from.object))
+            ,count(from.count)
+            ,impl(from.impl)
+            {
+               from.count = 0;
+            }
+
+            // No copy assignment, but allow move
+            in_flight& operator=(const in_flight&) = delete;
+            in_flight& operator=(in_flight&& from) {
+               object = std::move(from.object);
+               count = from.count;
+               impl = from.impl;
+               from.count = 0;
+            }
+
+            /**
+             * const accessor
+             * @return const reference to the contained object
+             */
+            const T& operator* () const {
+               return object;
+            }
+
+            /**
+             * mutable accessor (can be moved frmo)
+             * @return mutable reference to the contained object
+             */
+            T& operator* () {
+               return object;
+            }
```

Comment:
What is the point of these `operator*` ? You can just `std::move` the object without them which you actually do.